### PR TITLE
Added split and segment synching from json file to memory storage

### PR DESCRIPTION
**Pull request by chillaq**
# Python SDK

## Extra Notes

### CODE REVIEW

File: splitio/sync/split.py [hunk 1]
```diff
@@ -150,21 +153,30 @@ def kill_split(self, split_name, default_treatment, change_number):
         """
         self._split_storage.kill_locally(split_name, default_treatment, change_number)
 
+class LocalhostMode(Enum):
+    """types for localhost modes"""
+    LEGACY_YAML = 0
```
Comment (mmelograno) [inline splitio/sync/split.py:158]: LEGACY = 0
YAML = 1
JSON = 2

File: splitio/sync/segment.py [hunk 2]
```diff
@@ -174,13 +176,124 @@ def synchronize_segments(self, segment_names = None, dont_wait = False):
...
+        :rtype: bool
+        """
+        return self._segment_storage.get(segment_name) != None
+
+class LocalSegmentSynchronizer(object):
+    def __init__(self, segment_folder, split_storage, segment_storage):
+        """
+        Class constructor.
+
+        :param segment_api: API to retrieve segments from backend.
```
Comment (mmelograno) [inline splitio/sync/segment.py:203]: wrong doc param

File: splitio/sync/segment.py [hunk 3]
```diff
@@ -174,13 +176,124 @@ def synchronize_segments(self, segment_names = None, dont_wait = False):
...
+        :rtype: bool
+        """
+        try:
+            fetched = self._read_segment_from_json_file(segment_name)
+            if not self.segment_exist_in_storage(segment_name):
+                    self._segment_storage.put(segments.from_raw(fetched))
+                    self._segment_storage.set_change_number(segment_name, self._get_sha(json.dumps(fetched)))
+                    _LOGGER.debug("segment %s is added to storage", segment_name)
+            else:
+                if self._segment_storage.get_change_number(segment_name) != self._get_sha(json.dumps(fetched)):
```
Comment (mmelograno) [inline splitio/sync/segment.py:237]: are you comparing changeNumber against sha? you can track last sha in a dict in this class

Comment (mmelograno) [reply splitio/sync/segment.py:237]: changenumber is not a number from sha, is the last time the definition was updated

File: splitio/sync/split.py [hunk 4]
```diff
@@ -150,21 +153,30 @@ def kill_split(self, split_name, default_treatment, change_number):
...
+    def __init__(self, filename, split_storage, localhost_mode=LocalhostMode.LEGACY_YAML):
         """
         Class constructor.
 
         :param filename: File to parse splits from.
         :type filename: str
         :param split_storage: Split Storage.
         :type split_storage: splitio.storage.InMemorySplitStorage
+        :param localhost_mode: mode for localhost either JSON or YAML.
+        :type split_storage: splitio.storage.InMemorySplitStorage
```
Comment (mmelograno) [inline splitio/sync/split.py:174]: wrong doc param

File: splitio/sync/split.py [hunk 5]
```diff
@@ -150,21 +153,30 @@ def kill_split(self, split_name, default_treatment, change_number):
...
+
+    def __init__(self, filename, split_storage, localhost_mode=LocalhostMode.LEGACY_YAML):
         """
         Class constructor.
 
         :param filename: File to parse splits from.
         :type filename: str
         :param split_storage: Split Storage.
         :type split_storage: splitio.storage.InMemorySplitStorage
+        :param localhost_mode: mode for localhost either JSON or YAML.
```
Comment (mmelograno) [inline splitio/sync/split.py:173]: you won't pass this, you will determine which type is by the extension provided on the filename

File: splitio/sync/split.py [hunk 6]
```diff
@@ -318,3 +337,54 @@ def synchronize_splits(self, till=None):  # pylint:disable=unused-argument
...
+        return None
+
+    def _synchronize_json(self):
+        """Update splits in storage for json mode."""
+        if not self._filename.lower().endswith(('.json')):
+            raise ValueError("json File provided %s does not have .json extension." % self._filename)
+
+        fetched = self._read_splits_from_json_file(self._filename)
+        segment_list = set()
+        if self._current_json_sha == "-1" or self._get_sha(json.dumps(fetched)) != self._current_json_sha:
```
Comment (mmelograno) [inline splitio/sync/split.py:350]: same as segments, changenumber is timestamp, sha should be something that you track inside this instance

**Review by mmelograno — CHANGES_REQUESTED** [review]

File: splitio/sync/split.py (hunk 1 above)
Comment (chillaq) [reply splitio/sync/split.py:158]: will do

File: splitio/sync/segment.py (hunk 2 above)
Comment (chillaq) [reply splitio/sync/segment.py:203]: fixed

File: splitio/sync/segment.py (hunk 3 above)
Comment (chillaq) [reply splitio/sync/segment.py:237]: corrected to use only change number

File: splitio/sync/split.py (hunk 4 above)
Comment (chillaq) [reply splitio/sync/split.py:174]: fixed

File: splitio/sync/split.py (hunk 6 above)
Comment (chillaq) [reply splitio/sync/split.py:350]: added check for split changeNumber in addition to sha

File: splitio/sync/synchronizer.py [hunk 7]
```diff
@@ -506,39 +506,65 @@ def __init__(self, split_synchronizers, split_tasks):
...
-    def sync_all(self, max_retry_attempts=-1):
+    def sync_all(self, till=None):
         """
         Synchronize all splits.
-        
-        :param max_retry_attempts: Not used, added for compatibility
         """
         try:
-            self._split_synchronizers.split_sync.synchronize_splits(None)
+            return self.synchronize_splits()
```
Comment (mmelograno) [inline splitio/sync/synchronizer.py:514]: where is the segment logic?

File: splitio/sync/synchronizer.py [hunk 8]
```diff
@@ -307,7 +307,7 @@ def synchronize_splits(self, till, sync_segments=True):
     def sync_all(self, max_retry_attempts=_SYNC_ALL_NO_RETRIES):
         """
         Synchronize all splits.
-        
+
```
Comment (mmelograno) [inline splitio/sync/synchronizer.py:310]: you changed the class Synchronizer(BaseSynchronizer): contract

**Review by mmelograno — CHANGES_REQUESTED** [review]

**Review by mmelograno — APPROVED** [review]

**Pull request merged**